data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/accumulator/BufferTypeOptions.java

```diff
@@ -15,7 +18,16 @@
  */
 public enum BufferTypeOptions {
 
-    INMEMORY("in_memory", new InMemoryBuffer());
+    INMEMORY("in_memory", new InMemoryBuffer()),
+    LOCALFILE("local_file", new Object() {
```

Comment:
Why did you change to make this anonymous class? I think you can continue to make this work with `new LocalFileBufferFactory`. This helps with unit testing.

Reply:
Modified with `new LocalFileBufferFactory`.